sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -36,4 +60,133 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
       })
       .getMany();
   }
+
+  /**
+   * Save scholastic standing and create new assessment
+   * for an application.
+   * @param locationId location id.
+   * @param applicationId application id.
+   * @param auditUserId user that should be considered the one that is
+   * causing the changes.
+   * @param scholasticStandingData scholastic standing data to be saved.
+   */
+  async saveScholasticStandingCreateReassessment(
+    locationId: number,
+    applicationId: number,
+    auditUserId: number,
+    scholasticStandingData: ScholasticStanding,
+  ): Promise<StudentScholasticStanding> {
+    const application = await this.applicationRepo
+      .createQueryBuilder("application")
+      .select(["application", "currentAssessment.id", "offering.id"])
+      .where("application.id = :applicationId", { applicationId })
+      .where("location.id = :locationId", { locationId })
+      .innerJoin("application.currentAssessment", "currentAssessment")
+      .innerJoin("currentAssessment.offering", "offering")
+      .innerJoin("application.location", "location")
+      .getOne();
+
+    if (!application) {
+      throw new CustomNamedError(
+        "Application Not found or invalid current assessment or offering.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (application.applicationStatus !== ApplicationStatus.completed) {
+      throw new CustomNamedError(
+        "Cannot report a change for application with status other than completed.",
+        INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+      );
+    }
+    // Check if any assessment already in progress for this application.
+    await this.studentAssessmentService.assertAllAssessmentsCompleted(
+      application.id,
+    );
+
+    return this.connection.transaction(async (transactionalEntityManager) => {
+      const now = new Date();
+      const auditUser = { id: auditUserId } as User;
+
+      // Get existing offering.
+      const existingOffering = await this.offeringRepo
+        .createQueryBuilder("offering")
+        .select(["offering", "educationProgram.id", "institutionLocation.id"])
+        .where("offering.id = :offeringId", {
+          offeringId: application.currentAssessment.offering.id,
+        })
+        .innerJoin("offering.educationProgram", "educationProgram")
+        .innerJoin("offering.institutionLocation", "institutionLocation")
+        .getOne();
+
+      // Cloning existing offering.
+      const offering: EducationProgramOffering = { ...existingOffering };
+
+      // Deleting the id, from the cloned object.
+      // So that when its save its considered as a new EducationProgramOffering object.
+      delete offering.id;
+
+      const newStudyEndDate =
+        scholasticStandingData.dateOfChange ||
```

Comment:
I believe that this one should use ?? instead of ||.
The OR(||) will use the "right" value if the first one is "0"(zero) for instance, right?
What we really want here is to use the "right" value if the first one is null or undefined, hence the ?? would be better.

Reply:
i tried this before, it was not giving the excepted result, 
![image](https://user-images.githubusercontent.com/77353155/164092688-6753e9f5-4c40-4928-b310-5a4bad74748e.png)